Doc link fix

chriselion opened the pull request:
### Proposed change(s)
Make all links that were pointing to 0.15.1 or master in C# point to the `release_1_docs` tag.
Will re-audit this (including for missing links) after https://github.com/Unity-Technologies/ml-agents/pull/3864/ is merged).

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)



### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### Other comments

mmattar commented on com.unity.ml-agents/Runtime/Academy.cs:
```diff
@@ -19,7 +19,7 @@
  * set-up a learning environment and train the behavior of characters in a
  * Unity scene, please browse our documentation pages on GitHub:
- * https://github.com/Unity-Technologies/ml-agents/blob/0.15.1/docs/
+ * https://github.com/Unity-Technologies/ml-agents/tree/0.15.1/docs/
```
`0.15.1` --> `release_1_docs`

vincentpierre commented:
I think there are still some references to master and latest_release. do we want to remove all of them ?

```
***ml-agents/README.md:
   64: | **master (unstable)** | -- | [source](https://github.com/Unity-Technologies/ml-agents/tree/master) | [docs](https://github.com/Unity-Technologies/ml-agents/tree/master/docs/Readme.md) | [download](https://github.com/Unity-Technologies/ml-agents/archive/master.zip) |

***ml-agents/ml-agents/README.md:
   13:   [here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Python-API.md)
   19:   [here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Training-ML-Agents.md)
   33: [ML-Agents Toolkit documentation.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Readme.md)

***ml-agents/ml-agents-envs/README.md:
   14:   [here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Python-API.md)
   28: [ML-Agents Toolkit documentation.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Readme.md)


***ml-agents/README.md:
    6: [![docs badge](https://img.shields.io/badge/docs-reference-blue.svg)](https://github.com/Unity-Technologies/ml-agents/tree/latest_release/docs/)
    9: ([latest release](https://github.com/Unity-Technologies/ml-agents/releases/tag/latest_release))
   48: [here](https://github.com/Unity-Technologies/ml-agents/tree/latest_release/docs/Readme.md) to

 ***ml-agents/com.unity.ml-agents/Documentation~/com.unity.ml-agents.md:
  108: [installation instructions]: https://github.com/Unity-Technologies/ml-agents/blob/latest_release/docs/Installation.md

***ml-agents/docs/Installation-Anaconda-Windows.md:
  134: [here](https://github.com/Unity-Technologies/ml-agents/archive/latest_release.zip).

***ml-agents/ml-agents-envs/mlagents_envs/environment.py:
   80:             f"Please go to https://github.com/Unity-Technologies/ml-agents/releases/tag/latest_release "
```

chriselion commented:
Thanks @vincentpierre 
```
***ml-agents/README.md:
   64: | **master (unstable)** | -- | [source](https://github.com/Unity-Technologies/ml-agents/tree/master) | [docs](https://github.com/Unity-Technologies/ml-agents/tree/master/docs/Readme.md) | [download](https://github.com/Unity-Technologies/ml-agents/archive/master.zip) |
```
Leaving this one, it's an explicit link to master.

```
***ml-agents/ml-agents/README.md:
   13:   [here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Python-API.md)
   19:   [here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Training-ML-Agents.md)
   33: [ML-Agents Toolkit documentation.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Readme.md)

***ml-agents/ml-agents-envs/README.md:
   14:   [here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Python-API.md)
   28: [ML-Agents Toolkit documentation.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Readme.md)
```
Updated these to be relative links.

```
***ml-agents/README.md:
    6: [![docs badge](https://img.shields.io/badge/docs-reference-blue.svg)](https://github.com/Unity-Technologies/ml-agents/tree/latest_release/docs/)
    9: ([latest release](https://github.com/Unity-Technologies/ml-agents/releases/tag/latest_release))
   48: [here](https://github.com/Unity-Technologies/ml-agents/tree/latest_release/docs/Readme.md) to
```
Will tackle these in another PR when I update the table.
```
 ***ml-agents/com.unity.ml-agents/Documentation~/com.unity.ml-agents.md:
  108: [installation instructions]: https://github.com/Unity-Technologies/ml-agents/blob/latest_release/docs/Installation.md
```
Updated to `release_1_docs`

```
***ml-agents/docs/Installation-Anaconda-Windows.md:
  134: [here](https://github.com/Unity-Technologies/ml-agents/archive/latest_release.zip).
```
Reworded to link to https://github.com/Unity-Technologies/ml-agents/releases

```
***ml-agents/ml-agents-envs/mlagents_envs/environment.py:
   80:             f"Please go to https://github.com/Unity-Technologies/ml-agents/releases/tag/latest_release "
```
Reworded to link to https://github.com/Unity-Technologies/ml-agents/releases (same as other log links in that file).
